### PR TITLE
Enable integration test web application

### PR DESCRIPTION
**Pull request by JunielKatarn**
## Overview

See `vnext/TestWebSite/README.md`.

Introduces an integration test web site implemented as an ASP.NET Core application.

The web site allows serving any arbitrary HTTP response or WebSocket endpoint to validate networking functionality.

For local development, it can be started via `dotnet run`.
For continuous integration, the site gets deployed to IIS, removing the complexity of manual process management.

## Why
Web-related integration tests have the following issues:
- Hard to manage (manually starting processes and tracking IDs in CI).
- Complex to write (in-process C++ test servers are too limited for writing complex workflows).
- Limited functionality (node and in-process servers lack sophisticated Web features such as Origin Policy, static content, caching...).

## What
- Define .NET project `Microsoft.ReactNative.Test.Website`.
- Update the Desktop CI pipelines to deploy the website to an IIS server.
  - For this initial deployment, additional dependencies are downloaded and installed during PR validation.\
    A follow-up PR will update the CI agent images to include all dependencies.
  - Installs the following components/packages:
    - Internet Information Services (IIS)
    - [ASP.NET Core Hosting Bundle](https://dotnet.microsoft.com/permalink/dotnetcore-current-windows-runtime-bundle-installer).

See `vnext/TestWebSite/Program.cs`.
- The website uses ports `5555` (`http`, `ws`) and `5543` (`https`, `wss`).
- Tests can be mapped with an arbitrary web path to any lambda or concrete function as long as it:
  a. Receives an `HttpContext` argument.
  b. Returns `void` (synchronously) or `Task` (asynchronously).

## Caveats
- Temporarily disables WebSocket tests (they fail in CI runs, but work fine locally either using a .NET stand-alone process or IIS deployment).
- Only a handful of tests are migrated to the web site in this pull request.\
  The remaining tests will be progressively ported in subsequent PRs.
- This PR will **temporarily** add about 180 seconds to the PR critical path (Desktop).\
  This will change once the build agent images get updated to include the additional software.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12714)

### CODE REVIEW

File: vnext/TestWebSite/Facebook/React/Test/RNTesterIntegrationTests.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Net.WebSockets;
+using System.Text;
+
+namespace Facebook.React.Test
```
Comment (vmoroz) [inline vnext/TestWebSite/Facebook/React/Test/RNTesterIntegrationTests.cs:5]: Why is it Facebook? You wrote the code, right?

File: vnext/TestWebSite/localhost.cicert
Comment (vmoroz) [inline vnext/TestWebSite/localhost.cicert:1]: Is it required to add this file. I just wonder if there are any security risks.

Comment (JunielKatarn) [reply vnext/TestWebSite/localhost.cicert:1]: @dannyvv this file is only needed for CI/PR validation.
It is self-signed and not used anywhere else.
Is it OK to leave it checked in, or should we put it in a key vault and download as needed? 

Note, for local development, the SDK-included .NET certificates are used instead (see `vnext/TestWebsite/README.md`).

File: vnext/Desktop.IntegrationTests/HttpOriginPolicyIntegrationTest.cpp [hunk 2]
```diff
@@ -743,6 +743,46 @@ TEST_CLASS(HttpOriginPolicyIntegrationTest)
     TestOriginPolicy(serverArgs, clientArgs, s_shouldFail);
   } // FullCors304ForSimpleGetFails
 
+  TEST_METHOD(OfficeDev_OfficeJS_4144)
+  {
+    SetRuntimeOptionString("Http.GlobalOrigin", "http://orig.in");
```
Comment (NikoAri) [inline vnext/Desktop.IntegrationTests/HttpOriginPolicyIntegrationTest.cpp:748]: Why this has hardcoded "orig.in" url?

**Review by NikoAri — APPROVED** [review]

File: vnext/TestWebSite/localhost.cicert
Comment (JunielKatarn) [reply vnext/TestWebSite/localhost.cicert:1]: Generating in-place.

File: vnext/Desktop.IntegrationTests/HttpOriginPolicyIntegrationTest.cpp (hunk 2 above)
Comment (JunielKatarn) [reply vnext/Desktop.IntegrationTests/HttpOriginPolicyIntegrationTest.cpp:748]: No strict reason. Just chose an URL that would suggest its intended purpose.
For these type of tests, only syntactic validation is done. These `Origin` URLs are never connected to.

Suggestions for less confusing test URLs are welcome.

File: .ado/jobs/desktop.yml [hunk 3]
```diff
@@ -71,19 +71,22 @@ jobs:
 
           variables:
             - template: ../variables/windows.yml
+            - group: RNW Secrets
```
Comment (JunielKatarn) [inline .ado/jobs/desktop.yml:74]: TODO: Remove. No longer needed.

**Comment by JunielKatarn** [comment]
Reverting to draft.

Dealing with certificate installation.

**Comment by JunielKatarn** [comment]
Ready for merge.

**Pull request merged**